File: SQL-AI/ProgrammingLanguage/Leetcode-Implementation/Ruby/merge_sorted_array.rb
# @param {Integer[]} nums1
# @param {Integer} m
# @param {Integer[]} nums2
# @param {Integer} n
# @return {Void} Do not return anything, modify nums1 in-place instead.
def merge(nums1, m, nums2, n)
  # Merge two arrays from back to front
  # so avoid creating a new array
  first, second = m - 1, n - 1
  iter = m + n -1
  until iter < 0
    case
    when first < 0
      nums1[iter] = nums2[second]
      second -= 1
    when second < 0
      nums1[iter] = nums1[first]
      first -= 1
    when nums1[first] < nums2[second]
      nums1[iter] = nums2[second]
      second -= 1
    else
      nums1[iter] = nums1[first]
      first -= 1
    end
    iter -= 1
  end
end

num1, num2 = [1, 5, 7, 9], [3, 6, 10]
merge(num1, num1.length, num2, num2.length)
puts num1.inspect

File: SQL-AI/ProgrammingLanguage/Leetcode-Implementation/Python/MergeSortedArray.py
class Solution:
    # @param A  a list of integers
    # @param m  an integer, length of A
    # @param B  a list of integers
    # @param n  an integer, length of B
    # @return nothing
    def merge(self, A, m, B, n):
        a, b = 0, 0
        while a < m and b < n:
            if A[a] < B[b]:
                a += 1
            else:
                A.insert(a, B[b] )
                a, b = a + 1, b + 1
                m += 1
        while b != len(B):
            A[a] = B[b]
            a, b = a +1, b +1

if __name__ == "__main__":
    inst = Solution()
    A = [0]
    B = [1]
    inst.merge(A, 0, B, 1)
    print A
